tldw_Server_API/app/core/Prototype_Workspaces/service.py

```
@@ -0,0 +1,453 @@
+
+    async def _assert_branch_actor_active(
+        self,
+        *,
+        actor_type: str,
+        actor_shared_actor_id: str | None,
+    ) -> None:
+        if str(actor_type or "").strip().lower() != "external_collaborator":
+            return
+        actor = await self._repo.get_shared_actor(str(actor_shared_actor_id or ""))
+        if not actor or actor.get("is_revoked"):
+            raise RuntimeError("revoked shared actor cannot create or reuse branch sessions")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)